lnrpc/invoicesrpc/addinvoice.go

```diff
@@ -435,7 +456,11 @@ func selectHopHints(amtMSat lnwire.MilliSatoshi, cfg *AddInvoiceConfig,
 			zpay32.RouteHint([]zpay32.HopHint{hint}),
 		)
 
-		numHints++
+		// As we've just added a new hop hint, we'll accumulate it's
+		// available balance now to update our tally.
+		//
+		// TODO(roasbeef): have a cut off based on min bandwidth?
```

Comment:
We could still include some hints if the minimum isn't reached in the event that balances change between issuing the invoice and it being paid.

Reply:
I'm considering always adding the largest channel that can satisfy the payment, which should cover the non-mpp senders. This also can be randomized somewhat, but I need to check some details w.r.t ordering across instantiations of a map with the same contents. 

By min cut off, here I mean if we should exclude stuff like channels that only have a few hundred satoshis available as inbound.

Reply:
Tacked this on in the latests commit.